README.md

```diff
@@ -262,6 +262,40 @@ In `rv32emu` repository, there are some prebuilt ELF files for testing purpose.
 * `scimark2.elf` : See [tests/scimark2](tests/scimark2) [RV32MF]
 * `stream.elf` : See [tests/stream](tests/stream.c) [RV32MF]
 
+## Benchmarks
+The benchmarks ([Source code](https://github.com/sysprog21/rv32emu/tree/master/tests)) are classified into three categories based on their characteristics:
+### 1. Computing Intensive Benchmarks:
+* puzzle: The N-puzzle is a sliding puzzle where numbered
+square tiles are arranged randomly with one tile missing,
+designed for solving the N-puzzle problem.
+* Pi: Calculates the millionth digit of π.
+* miniz: Compresses and decompresses 8 MiB of data.
+* primes: Finds the largest prime number below 33333333.
+* sha512: Computes the SHA-512 hash of 64 MiB of data.
+### 2. I/O Intensive Benchmarks:
+* Richards: An OS task scheduler simulation benchmark
+for comparing system implementations.
+* Dhrystone: Evaluates string operations, involves frequent
+memory I/O, and generates the performance metric.
+### 3. Computing and I/O Hybrid Benchmarks:
+* Mandelbrot: A benchmark based on the Mandelbrot set,
+which uses fixed-point arithmetic and involves numerous
+integer operations.
+* AES: Includes 23 encryption and decryption algorithms
+adhering to the Advanced Encryption Standard.
+* Nqueens: A puzzle benchmark where n queens are placed
+on an n × n chessboard without attacking each other,
+using deep recursion for execution.
+* qsort: Sorts an array with 50 million items.
+
+These Benchmarks performed by rv32emu (interpreter-only) and Spike v1.1.0. Ran on Intel Core i7-11700 CPU running at 2.5 GHz and an Ampere eMAG 8180
+microprocessor equipped with 32 Arm64 cores, capable of speeds up to 3.3 GHz. Both systems ran Ubuntu Linux 22.04.1 LTS. We utilized gcc version 12.3, configured as riscv32-unknown-elf-gcc.
+
+_x86-64_
+![](docs/interp-bench-x86.png)
+
+_ARM64_
```

Comment:
Always use "Arm64" instead of "ARM64" as [Arm Trademark Use Guidelines](https://www.arm.com/zh-TW/company/policies/trademarks/guidelines-trademarks) suggests.
> Do not alter the spelling or form of Arm's trademarks by abbreviating them, creating acronyms, translating them, joining them to other words, symbols or numbers (either as one word or with a hyphen - unless otherwise permitted, e.g. Arm® Cortex®-A15), or using improper capitalization.